containerd-http-proxy/configure_http_proxy.yaml

```diff
@@ -0,0 +1,94 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: containerd-http-proxy
+spec:
+  selector:
+    matchLabels:
+      name: containerd-http-proxy
+  template:
+    metadata:
+      labels:
+        name: containerd-http-proxy
+    spec:
+      hostPID: true
+      volumes:
+      - name: systemd-containerd-service
+        hostPath:
+          path: /etc/systemd/system/containerd.service.d
+          type: DirectoryOrCreate
+      initContainers:
+      - name: startup-script
+        image: gke.gcr.io/debian-base:bookworm-v1.0.0-gke.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: systemd-containerd-service
+          mountPath: /etc/systemd/system/containerd.service.d
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          set -u
+
+          validate_proxy() {
+            input_string=$1
```

Comment:
what kind of values can the input string be?
can we add a check that makes sure the match the regex of expected values?

Reply:
They should be urls. NO_PROXY could have multiple address separated by comma, and allow wildcard. Dash doesn't support using regex very well and I'm worried about scenarios where some valid cases could be rejected. And I think checking space and newline should be sufficient.